model/lifecycle.go

```diff
@@ -400,6 +409,10 @@ func RestartBuild(buildId string, taskIds []string, abortInProgress bool, caller
 		}
 	}
 
+	if err := build.UpdateActivation([]string{buildId}, true, caller); err != nil {
```

Comment:
Since much of this change is changing order, it would be helpful to include a comment so that someone doesn't inadvertently reintroduce this bug by changing the order back for some other reason.

Reply:
done